core/domain/stats_domain.py

```diff
@@ -560,23 +587,31 @@ def __init__(self, calculation_output_type):
 
 class AnswerFrequencyList(AnswerCalculationOutput):
     """Domain object that represents a list of AnswerOccurrences."""
-    def __init__(self, answer_occurrences=None):
+    def __init__(self, answer_occurrences=None, limit=None):
```

Comment:
"limit" is not a concept inherent to this list, don't introduce it here. It should not be a member property. If you need to trim the list then introduce a new method specifically for doing that (e.g. answer_frequency_list.truncate(limit)).

Reply:
Done, but I like using the limit for getting most_common back. Maybe I can introduce it as a private member?